studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { FC, Fragment, useState } from 'react'
+import { useRouter } from 'next/router'
+import { Button, IconBell, Popover } from '@supabase/ui'
+import { Notification, NotificationStatus } from '@supabase/shared-types/out/notifications'
+
+import { Project } from 'types'
+import { useNotifications, useStore } from 'hooks'
+import { patch, post } from 'lib/common/fetch'
+import { API_URL } from 'lib/constants'
+import NotificationRow from './NotificationRow'
+import ConfirmModal from 'components/ui/Dialogs/ConfirmDialog'
+
+interface Props {}
+
+const NotificationsPopover: FC<Props> = () => {
+  const router = useRouter()
+  const { app, ui } = useStore()
+  const { notifications, refresh } = useNotifications()
+
+  const [projectToRestart, setProjectToRestart] = useState<Project>()
+  const [targetNotification, setTargetNotification] = useState<Notification>()
+
+  if (!notifications) return <div />
+
+  const hasNewNotifications = notifications?.some(
+    (notification) => notification.notification_status === NotificationStatus.New
+  )
+
+  const onOpenChange = async (open: boolean) => {
+    if (!open) {
+      // Mark notifications as seen
+      const notificationIds = notifications
+        ?.filter((notification) => notification.notification_status === NotificationStatus.New)
```

Comment:
Super super tiny nit: `?.` isn't needed here because of the `!notifications` if above